parmed/gromacs/gromacsgro.py

```diff
@@ -253,17 +254,17 @@ def write(struct, dest, precision=3, nobox=False):
             a, b, c = reduce_box_vectors(*box_lengths_and_angles_to_vectors(
                             *struct.box))
             if all([abs(x-90) < TINY for x in struct.box[3:]]):
-                dest.write('%10.5f'*3 % (a[0]/10, b[1]/10, c[2]/10))
+                dest.write(boxfmt*3 % (a[0]/10.0, b[1]/10.0, c[2]/10.0))
```

Comment:
Looks sensible to me.

Nitpick: the `10.0` changes are not necessary when doing `from __future__ import division`. It will perform floating point division even when operating on integers.

Reply:
And what's more is that leaving it as 10 is marginally faster :smiley:

```python
In [1]:

def divide_10(number):
    return number / 10
​
def divide_10_0(number):
    return number / 10.0

In [2]:

%timeit divide_10(123456)
126 ns ± 2.88 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
In [3]:

%timeit divide_10_0(123456)
213 ns ± 2.76 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

The reason for this is that Python caches small integers at startup for performance reasons.  As a result, leaving them as 10 requires Python simply to fetch the cached integer, whereas for the latter Python has to create a new float for 10.0 (although it would use the same instance in all 3 places in that statement).

Here is a demo of that:

```python

In [8]:

x = 10
y = 10
a = 10.0
b = 10.0
In [9]:

x is y
Out[9]:
True
In [10]:

a is b
Out[10]:
False
```

Of course this is an unnoticeable cost, the discussion is purely academic :).  And for other Python implementations that may not cache small integers there may be no benefit. (e.g., pypy which ParmEd explicitly supports and provides speedups around 5x over CPython).

Reply:
> 10.0 changes are not necessary when doing from __future__ import division. It will perform floating point division even when operating on integers.

I would argue that it's more transparent to code maintainers that it is a unit conversion that is fundamentally a floating point conversion, not an integer conversion, even if one is guaranteed to get the same answer either way.

Reply:
And we can always multiply by 0.1 instead.  I wonder what that speed would be.

Reply:
> And we can always multiply by 0.1 instead. I wonder what that speed would be.

That would almost certainly be faster :)